asyncpg/connection.py

```diff
@@ -1667,6 +1696,20 @@ async def _execute(
             )
         return result
 
+    def logger(self, callback):
+        return _LoggingContext(self, callback)
+
+    def _log_query(self, query, args, timeout, elapsed):
+        if not self._query_loggers:
+            return
+        con_ref = self._unwrap()
+        record = LoggedQuery(query, args, timeout, elapsed)
+        for cb in self._query_loggers:
+            if cb.is_async:
+                self._loop.create_task(cb.cb(con_ref, record))
```

Comment:
Passing a `con_ref` is probably unnecessary?

Reply:
I likely wouldn't use it, so happy to remove it, but I put it there so you could potentially log queries by host.

Reply:
A concern here is potentially retaining references to free-d connections.  Other callbacks take it, of course, but that's an API decision I've come to regret.  Perhaps we can pass connection's `addr` and `params` instead?

Reply:
Works for me!